### DISCUSSION TITLE
Why the CPU emit lua to execute instead of executing premade function ?

### DISCUSSION THREAD

**Discussion by mms92** [wiremod/wire#2550] [Q&A] [answered]
everything is in the title ,
 
why does the wiremod team decided to make the cpu execute new , emitted lua program, instead of calling pre-established function, to execute the cpu program ?

**Comment by vurvdev** [comment, marked as answer]
Just barely saw this discussion,

For a real answer you'd have to ask the creator of the cpu (BlackFox on the discord).

But as for why you'd transpile a language in general, it's easier, and would run faster. I want E2 to transpile to lua at some point to run faster. No idea why *specifically* it was chosen to be done for the cpu though.
